seq_strstr/host/dpu_host.c

```diff
@@ -0,0 +1,162 @@
+#include <dpu.h>
+#include <dpu_log.h>
+#include <dpu_memory.h>
+#include <dpu_program.h>
+#include <assert.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <fcntl.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <time.h>
+#include <sys/time.h>
+
+#include "dpu_host.h"
+#include "common.h"
+// dpu binary location TBD
+#define DPU_BINARY "build/strstr_dpu"
+#define DPU_LOG_ENABLE 1
+#define WRITE_OUT 1
+#define ALIGN(_p, _width) (((unsigned int)_p + (_width-1)) & (0-_width))
+
+
+
+dpu_error_t dpu_copy_to_dpu(struct dpu_set_t dpu, const char *symbol_name, uint32_t symbol_offset, const void *src, size_t length){
+    dpu_error_t status;
+    struct dpu_program_t *program;
+    struct dpu_symbol_t symbol;
+  
+    program = dpu_get_program((dpu.dpu));
+    if ((status = dpu_get_symbol(program, symbol_name, &symbol)) != DPU_OK) {
+        return status;
+    }
+
+    return dpu_copy_to_symbol_dpu(dpu.dpu, symbol, symbol_offset, src, length);
+}
+
+
+dpu_error_t dpu_copy_from_dpu(struct dpu_set_t dpu, const char *symbol_name, uint32_t symbol_offset, void *dst, size_t length){
+    dpu_error_t status;
+    struct dpu_program_t *program;
+    struct dpu_symbol_t symbol;
+  
+    program = dpu_get_program((dpu.dpu));
+    if ((status = dpu_get_symbol(program, symbol_name, &symbol)) != DPU_OK) {
+        return status;
+    }
+
+    return dpu_copy_from_symbol_dpu(dpu.dpu, symbol, symbol_offset, dst, length);
+}
+
+void printRecord(char* record_start, uint32_t length) {
+    for(uint32_t i =0; i< length; i++){
+        printf("%c", record_start[i]);
+    }
+
+    printf("\n");
+    printf("\n");
+}
+
+
+
+#define ALIGN(_p, _width) (((unsigned int)_p + (_width-1)) & (0-_width))
```

Comment:
A third time! You really like this macro :-)

Reply:
I was a little lazy to change the makefile, gonna update them.  In addition, I did notice when _p is larger than an unsigned int (long), this macro becomes a bit problematic.